mapk_demon/receivers.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import division
+
+from functools import partial
+
+import numpy as np
+import pyro
+from pyro import sample
+from pyro.distributions import Normal, Uniform
+from torch import tensor
+from mapk_demon.transmitters import mapk_signaling
+from pyro.infer.mcmc import MCMC
+from pyro.infer.mcmc.nuts import NUTS
+from pyro.infer import EmpiricalMarginal
+from causal_demon.inference import infer_dist
+import torch
+import matplotlib.pyplot as plt
+
+def infer_mcmc(model):
+    nuts_kernel = NUTS(model, adapt_step_size=True)
+    return MCMC(nuts_kernel, num_samples=500, warmup_steps=300)
+
+
+def g1(a):
+    return a / (a + 1)
+
+
+def g2(a):
+    return a ** 2 / (a ** 2 + a + 1)
+
+
+def f(mu, N):
+    return N * mu ** (0.5) + mu
+
+
+# Parameters
+
+params = {
+        "alpha_3k": Normal(500, 1),
+        "alpha_2k": Normal(500, 1),
+        "alpha_k": Normal(500, 1),
+        "nu_3k": Normal(.15, .05),
+        "nu_2k": Normal(.15, .05),
+        "nu_k": Normal(.15, .05)
+}
+'''
+params = {
+    "alpha_3k": 500,
+    "alpha_2k": 500,
+    "alpha_k": 500,
+    "nu_3k": .15,
+    "nu_2k": .15,
+    "nu_k": .15
+}
+'''
+
+hyperparameters = {
+    "t_3k": 1.2,
+    "t_2k": 1.2,
+    "t_k": 0.003
+}
+
+
+def f_map3k(E1, N_3k, params):
+    total_3k = hyperparameters['t_3k']
+    alpha_3k = sample("alpha_3k",params['alpha_3k'])
```

Comment:
This will not work, if you consider the math, you will see that there are many combinations of `alpha_3k/nu_3k` that will provide the same value.  Create a new parameter `beta_3k = alpha_3k/nu_3k`, create a prior for `beta_3k` and sample that from a prior.  Focus your inference on `beta_3k`